### ISSUE TITLE
Missing output of ERROR in TAP file for `unittest.TestCase` code

### ISSUE THREAD

**Issue by andilabs** [python-tap/pytest-tap#39]
The _ ERROR at setup of test case (unittest.TestCase) but run with pytest and using pytest fixture are not present inside the tap file although are present in log of pytest; the one which resulted OK are in TAP file. Whe tests fails it is also present in TAP, but not when ther is any kind of error. 

I would expect errors are in tap file too. Is it correct assumption? If yest let me know and I try to provide minimal example to reproduce.

```
============================= test session starts ==============================
platform linux -- Python 3.5.2, pytest-5.0.1, py-1.8.0, pluggy-0.12.0 -- /nzutils/env/bin/python3
cachedir: .pytest_cache
rootdir: /nzutils/application/columbus, inifile: setup.cfg
plugins: cov-2.7.1, flake8-1.0.4, flask-0.15.0, forked-1.0.2, mccabe-1.0, openfiles-0.4.0, tap-2.2, xdist-1.29.0
collecting ... collected 36 items

nzutils/application/columbus/tests/test_api.py::test_api_get_list_users PASSED [  2%]
nzutils/application/columbus/tests/test_api.py::test_api_get_list_groups PASSED [  5%]
nzutils/application/columbus/tests/test_api.py::test_get_metadata_by_non_existing_identifier PASSED [  8%]
nzutils/application/columbus/tests/test_api.py::test_get_metadata_by_non_existing_table_path PASSED [ 11%]
nzutils/application/columbus/tests/test_api.py::test_get_metadata_by_existing_identifier PASSED [ 13%]
nzutils/application/columbus/tests/test_app.py::test_env PASSED          [ 16%]
nzutils/application/columbus/tests/test_app.py::test_app PASSED          [ 19%]
nzutils/application/columbus/tests/test_entity_groups.py::TestEntityGroups::testCountColumn PASSED [ 22%]
nzutils/application/columbus/tests/test_entity_groups.py::TestEntityGroups::testDeleted PASSED [ 25%]
nzutils/application/columbus/tests/test_entity_groups.py::TestEntityGroups::testLatestIngested PASSED [ 27%]
nzutils/application/columbus/tests/test_entity_groups.py::TestEntityGroups::testRowStats PASSED [ 30%]
nzutils/application/columbus/tests/test_index.py::test_homepage PASSED   [ 33%]
nzutils/application/columbus/tests/test_index.py::test_users_list PASSED [ 36%]
nzutils/application/columbus/tests/test_index.py::test_users_userinfo PASSED [ 38%]
nzutils/application/columbus/tests/test_index.py::test_entities_list_ordering_by_table_name PASSED [ 41%]
nzutils/application/columbus/tests/test_index.py::test_entities_list_ordering_by_table_schema PASSED [ 44%]
nzutils/application/columbus/tests/test_index.py::test_ordering_on_each_of_entity_view_list_columns_works PASSED [ 47%]
nzutils/application/columbus/tests/test_index.py::test_user_can_not_edit_entity_with_restricted_access PASSED [ 50%]
nzutils/application/columbus/tests/test_index.py::test_only_the_owner_can_edit_entity_with_restricted_access PASSED [ 52%]
nzutils/application/columbus/tests/test_index.py::test_user_with_user_access_can_edit_entity PASSED [ 55%]
nzutils/application/columbus/tests/test_index.py::test_user_without_required_group_membership_can_not_edit_entity PASSED [ 58%]
nzutils/application/columbus/tests/test_index.py::test_user_with_group_access_can_edit_entity PASSED [ 61%]
nzutils/application/columbus/tests/test_index.py::test_user_with_group_membership_can_edit_entity PASSED [ 63%]
nzutils/application/columbus/tests/test_models.py::test_env PASSED       [ 66%]
nzutils/application/columbus/tests/test_models.py::test_app PASSED       [ 69%]
nzutils/application/columbus/tests/test_models.py::test_extraction_in_display_table_name_is_correct PASSED [ 72%]
nzutils/application/columbus/tests/test_models.py::test_sorting_by_hybrid_property_display_table_name PASSED [ 75%]
nzutils/application/columbus/tests/test_models.py::UserGroupsTest::test_get_user_groups_for_newly_created_user PASSED [ 77%]
nzutils/application/columbus/tests/test_models.py::UserGroupsTest::test_get_user_groups_for_super_user PASSED [ 80%]
nzutils/application/columbus/tests/test_models.py::UserGroupsTest::test_get_user_groups_non_existing_user PASSED [ 83%]
nzutils/application/columbus/tests/test_models.py::UserGroupsTest::test_is_superuser PASSED [ 86%]
nzutils/application/columbus/tests/test_ui.py::TestEntityFilteringAndSearchingUI::test_all_filter_options_rendered ERROR [ 88%]
nzutils/application/columbus/tests/test_ui.py::TestEntityFilteringAndSearchingUI::test_filtering_by_rows ERROR [ 91%]
nzutils/application/columbus/tests/test_ui.py::TestEntityFilteringAndSearchingUI::test_filters_combined_with_free_text_search ERROR [ 94%]
nzutils/application/columbus/tests/test_ui.py::TestEntityFilteringAndSearchingUI::test_free_text_search_works ERROR [ 97%]
nzutils/application/columbus/tests/test_ui.py::TestEntityFilteringAndSearchingUI::test_perform_basic_ui_check ERROR [100%]```

**Comment by mblayman** [comment]
Hi, @andilabs, thanks for the report!

I don't fully understand the problem you're describing. It sounds like something is missing, but I'm not seeing the full picture. Can you please provide the TAP output and reference the particular line(s) that you would expect to see that you believe are missing?

**Comment by andilabs** [comment]
in TAP files I have only represented passed test wtih `ok....` and no representation of that with ERROR. I would expect for them lines with `not ok...`

**Comment by andilabs** [comment]
the `pytest --tap-files` output is:
```
============================= test session starts ==============================
platform linux -- Python 3.5.2, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: /nzutils/application/columbus, inifile: setup.cfg
plugins: cov-2.7.1, flake8-1.0.4, flask-0.15.0, forked-1.0.2, mccabe-1.0, openfiles-0.4.0, tap-2.2, xdist-1.29.0
collected 36 items

test_api.py .....                                                        [ 13%]
test_app.py ..                                                           [ 19%]
test_entity_groups.py ....                                               [ 30%]
test_index.py ............                                               [ 63%]
test_models.py ........                                                  [ 86%]
test_ui.py EEEEE                                                         [100%]

==================================== ERRORS ====================================
_ ERROR at setup of TestEntityFilteringAndSearchingUI.test_all_filter_options_rendered _

self = <test_ui.TestEntityFilteringAndSearchingUI testMethod=test_all_filter_options_rendered>
session = <sqlalchemy.orm.session.Session object at 0x7f5aef868cf8>

    @pytest.fixture(autouse=True)
    def test_filtering_by_rows(self, session):
        entity_1 = _create_entity(table_path='some_name1.xls', rows=3)
        entity_2 = _create_entity(table_path='another_name1.xls', rows=5)
        session.add(entity_1)
        session.add(entity_2)
        base_url = 'http://columbus:8080/entityview/list/'
>       self.driver.get(base_url)
E       AttributeError: 'NoneType' object has no attribute 'get'

test_ui.py:134: AttributeError
__ ERROR at setup of TestEntityFilteringAndSearchingUI.test_filtering_by_rows __

self = <test_ui.TestEntityFilteringAndSearchingUI testMethod=test_filtering_by_rows>
session = <sqlalchemy.orm.session.Session object at 0x7f5aef522518>

    @pytest.fixture(autouse=True)
    def test_filtering_by_rows(self, session):
        entity_1 = _create_entity(table_path='some_name1.xls', rows=3)
        entity_2 = _create_entity(table_path='another_name1.xls', rows=5)
        session.add(entity_1)
        session.add(entity_2)
        base_url = 'http://columbus:8080/entityview/list/'
>       self.driver.get(base_url)
E       AttributeError: 'NoneType' object has no attribute 'get'

test_ui.py:134: AttributeError
_ ERROR at setup of TestEntityFilteringAndSearchingUI.test_filters_combined_with_free_text_search _

self = <test_ui.TestEntityFilteringAndSearchingUI testMethod=test_filters_combined_with_free_text_search>
session = <sqlalchemy.orm.session.Session object at 0x7f5aef863d68>

    @pytest.fixture(autouse=True)
    def test_filtering_by_rows(self, session):
        entity_1 = _create_entity(table_path='some_name1.xls', rows=3)
        entity_2 = _create_entity(table_path='another_name1.xls', rows=5)
        session.add(entity_1)
        session.add(entity_2)
        base_url = 'http://columbus:8080/entityview/list/'
>       self.driver.get(base_url)
E       AttributeError: 'NoneType' object has no attribute 'get'

test_ui.py:134: AttributeError
_ ERROR at setup of TestEntityFilteringAndSearchingUI.test_free_text_search_works _

self = <test_ui.TestEntityFilteringAndSearchingUI testMethod=test_free_text_search_works>
session = <sqlalchemy.orm.session.Session object at 0x7f5aef50c860>

    @pytest.fixture(autouse=True)
    def test_filtering_by_rows(self, session):
        entity_1 = _create_entity(table_path='some_name1.xls', rows=3)
        entity_2 = _create_entity(table_path='another_name1.xls', rows=5)
        session.add(entity_1)
        session.add(entity_2)
        base_url = 'http://columbus:8080/entityview/list/'
>       self.driver.get(base_url)
E       AttributeError: 'NoneType' object has no attribute 'get'

test_ui.py:134: AttributeError
_ ERROR at setup of TestEntityFilteringAndSearchingUI.test_perform_basic_ui_check _

self = <test_ui.TestEntityFilteringAndSearchingUI testMethod=test_perform_basic_ui_check>
session = <sqlalchemy.orm.session.Session object at 0x7f5aef51f898>

    @pytest.fixture(autouse=True)
    def test_filtering_by_rows(self, session):
        entity_1 = _create_entity(table_path='some_name1.xls', rows=3)
        entity_2 = _create_entity(table_path='another_name1.xls', rows=5)
        session.add(entity_1)
        session.add(entity_2)
        base_url = 'http://columbus:8080/entityview/list/'
>       self.driver.get(base_url)
E       AttributeError: 'NoneType' object has no attribute 'get'

test_ui.py:134: AttributeError
===================== 31 passed, 5 error in 28.94 seconds ======================
```

there is no file generated for test_ui tests wich ended up with ERROR

**Comment by mblayman** [comment]
You might be right about pytest's behavior. In theory, the hooks that `pytest-tap` plug should collect that information and the plugin could act on that. Could you produce the simplest possible test case that can reproduce this behavior? I might have a chance to look at the hook behavior later if I have something to work against.

**Comment by andilabs** [comment]
here is the minimum exmaple: https://github.com/andilabs/pytest-tap-failure-demo

when run with `pytest --tap-files`
the output is:
```
(venv) akos@akos:/nfs/home/akos/Projects/pytest-tap-failure-demo$ pytest --tap-files                                                                                                                        
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.5.2, pytest-5.2.1, py-1.8.0, pluggy-0.13.0                                                                                                                                       
rootdir: /nfs/home/akos/Projects/pytest-tap-failure-demo                                                                                                                                                    
plugins: cov-2.8.1, flake8-1.0.4, flask-0.15.0, forked-1.1.1, mccabe-1.0, openfiles-0.4.0, tap-2.3, xdist-1.30.0                                                                                            
collected 2 items                                                                                                                                                                                           
                                                                                                                                                                                                            
test_demo.py FE                                                                                                                                                                                      [100%] 
                                                                                                                                                                                                            
================================================================================================== ERRORS ==================================================================================================
____________________________________________________________________________________ ERROR at setup of test_basic_error ____________________________________________________________________________________
                                                                                                                                                                                                            
    @pytest.fixture                                                                                                                                                                                         
    def some_fixture():                                                                                                                                                                                     
>       assert 2 == 3                                                                                                                                                                                       
E       assert 2 == 3                                                                                                                                                                                       
                                                                                                                                                                                                            
conftest.py:7: AssertionError                                                                                                                                                                               
================================================================================================= FAILURES =================================================================================================
_____________________________________________________________________________________________ test_basic_fail ______________________________________________________________________________________________
                                                                                                                                                                                                            
    def test_basic_fail():                                                                                                                                                                                  
>       assert 2 == 3                                                                                                                                                                                       
E       assert 2 == 3                                                                                                                                                                                       
                                                                                                                                                                                                            
test_demo.py:3: AssertionError                                                                                                                                                                              
======================================================================================== 1 failed, 1 error in 0.31s ========================================================================================
(venv) akos@akos:/nfs/home/akos/Projects/pytest-tap-failure-demo$                                                                                                                                           

in short - tests:
```
def test_basic_fail():
    assert 2 == 3


def test_basic_error(some_fixture):
    assert 2 == 2

```

test fixture:
```

import pytest



@pytest.fixture
def some_fixture():
    assert 2 == 3
```
```
but the generated TAP file contains only this:
(venv) akos@akos:/nfs/home/akos/Projects/pytest-tap-failure-demo$ cat test_demo.py.tap
```
# TAP results for test_demo.py
not ok 1 test_demo.py::test_basic_fail
#
#     def test_basic_fail():
# >       assert 2 == 3
# E       assert 2 == 3
#
# test_demo.py:3: AssertionError
1..1
```

**Comment by mblayman** [comment]
@andilabs Thanks for your help with this. I fixed the issue. There are a couple of other things that I would like to do on this repo, but I should be able to put out a 2.4 release in a week or two.
